Improve the specification of the SOAP module (#117)
ballerina-platform/module-ballerina-soap: docs/spec/spec.md

# Specification: Ballerina SOAP Library

> _Owners_: @shafreenAnfar @MadhukaHarith92  

Reviewer: We should add your name here.

Author: Fixed in [0582c3e](https://github.com/ballerina-platform/module-ballerina-soap/pull/117/commits/0582c3eb0e040ac6129a04c88fb6b873ee97c791)

> _Reviewers_: @shafreenAnfar  

Reviewer: Better to add line breaks after each line in this section.

```suggestion
_Owners_: @shafreenAnfar @MadhukaHarith92 \
```

_Created_: 2023/06/07  
_Updated_: 2023/12/01
_Edition_: Swan Lake

## Introduction

> This is the specification for the SOAP standard library of [Ballerina language](https://ballerina.io/), which provides APIs to send an ordinary XML request to a SOAP backend by specifying the necessary details to construct a SOAP envelope.

Reviewer: ```suggestion
This is the specification for the SOAP package of [Ballerina language](https://ballerina.io/), which provides APIs to send an ordinary XML request to a SOAP backend by specifying the necessary details to construct a SOAP envelope.
```

> The SOAP library specification has evolved and may continue to evolve in the future. The released versions of the specification can be found under the relevant GitHub tag.

Reviewer: ```suggestion
The SOAP package specification has evolved and may continue to evolve in the future. The released versions of the specification can be found under the relevant GitHub tag.
```

Author: Fixed in [0582c3e](https://github.com/ballerina-platform/module-ballerina-soap/pull/117/commits/0582c3eb0e040ac6129a04c88fb6b873ee97c791)

> If you have any feedback or suggestions about the library, start a discussion via a [GitHub issue](https://github.com/ballerina-platform/ballerina-library/issues) or in the [Discord server](https://discord.gg/ballerinalang). Based on the outcome of the discussion, the specification and implementation can be updated. Community feedback is always welcome. Any accepted proposal, which affects the specification is stored under `/docs/proposals`. Proposals under discussion can be found with the label `type/proposal` in GitHub.

Reviewer: Check similar occurrences. The convention is this:

- Library is an umbrella term. The Ballerina library includes all the Ballerina packages and connectors.
- Package is the single entity we publish

```suggestion
If you have any feedback or suggestions about the package, start a discussion via a [GitHub issue](https://github.com/ballerina-platform/ballerina-library/issues) or in the [Discord server](https://discord.gg/ballerinalang). Based on the outcome of the discussion, the specification and implementation can be updated. Community feedback is always welcome. Any accepted proposal, which affects the specification is stored under `/docs/proposals`. Proposals under discussion can be found with the label `type/proposal` in GitHub.
```

Author: Fixed in [0582c3e](https://github.com/ballerina-platform/module-ballerina-soap/pull/117/commits/0582c3eb0e040ac6129a04c88fb6b873ee97c791)
